servers/physics_2d/shape_2d_sw.cpp

```diff
@@ -318,7 +313,9 @@ bool CircleShape2DSW::intersect_segment(const Vector2 &p_begin, const Vector2 &p
 
 real_t CircleShape2DSW::get_moment_of_inertia(real_t p_mass, const Size2 &p_scale) const {
 
-	return (radius * radius) * (p_scale.x * 0.5 + p_scale.y * 0.5);
+	real_t a = radius * p_scale.x;
+	real_t b = radius * p_scale.y;
+	return p_mass * (a * a + b * b) / 4;
```

Comment:
Wouldn't `a * a` be the same as `radius * radius * p_scale.x * p_scale.x` then? And `radius * radius` will also happen for `b`. Might be more efficient to calculate that only once with `real_t r2 = radius * radius;`

Reply:
Another way is written as `return p_mass * (radius * radius) * p_scale.dot(p_scale) / 4;`. Which needs same calculation times in fact. And I think the origin code looks more easy to understand.